packages/dcos-integration-test/extra/conftest.py

```diff
@@ -177,6 +177,15 @@ def _wait_for_srouter_slaves_endpoints(self):
             assert "id" in data
             assert data["id"] == slave_id
 
+    @retrying.retry(wait_fixed=2000,
+                    retry_on_result=lambda r: r is False,
+                    retry_on_exception=lambda _: False)
+    def _wait_for_metronome(self):
+        r = self.get('/service/metronome/v1/jobs')
+        if r.status_code == 500 or r.status_code == 504:
```

Comment:
Can we get a comment as to why 500 and 504 we should retry on, but on all other error codes we shouldn't (I'm a fan of only waiting for the codes we know happen from adminrouter before it comes up, but this code doesn't give any explanation currently why these ones)